### PR TITLE
Feature/mainpage

### PR DESCRIPTION
**Pull request by Marin-de-Drouas**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,10 +59,13 @@
...
   "dependencies": {
+    "@balkangraph/familytree.js": "^1.7.28",
     "express": "^4.18.2",
     "react": "18.2.0",
+    "react-d3-tree": "^3.5.1",
     "react-dom": "18.2.0",
     "react-router-dom": "^6.10.0",
-    "tslib": "^2.3.0"
+    "tslib": "^2.3.0",
+    "uninstall": "^0.0.0"
```
Comment (EtienneDx) [inline package.json]: this needs to go ^^'

File: packages/front-end/src/app/app.tsx [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+    <BrowserRouter>
+      <Routes>
+        <Route path='/'>
+          <Route index element={<FamilyTreeDisplay />} />
+          <Route path='/add-person' element={<AddPerson />} />
+          <Route path='/login' element={<Login />} />
+          <Route path='/register' element={<Register />} />
+        </Route>
+      </Routes>
+    </BrowserRouter>
```
Comment (EtienneDx) [inline packages/front-end/src/app/app.tsx:18]: Just wondering about front-end BDD's cucumber integration, I think the router element needs to go outside of App (and into index) for it to work properly. Altough considering the attention we've given to front-end testing, not a major issue

File: packages/front-end/src/app/family-tree-page.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      id: 5,
+      mid: 1,
+      fid: 2,
+      name: 'Emma Stevens',
+      gender: 'female',
+      img: 'https://cdn.balkan.app/shared/w10/3.jpg',
+    },
+  ];
+
+function BarMenu(): JSX.Element {
```
Comment (EtienneDx) [inline packages/front-end/src/app/family-tree-page.tsx]: nit: This is a component in itself, could be moved to another file

File: packages/front-end/src/app/login-page.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      try {
+        const authToken = localStorage.getItem('authToken');
+        const response = await fetch('/login', {
+          method: 'POST',
+          headers: { 
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${authToken}`,
```
Comment (EtienneDx) [inline packages/front-end/src/app/login-page.tsx]: When we login, we shouldn't have a token yet

File: packages/front-end/src/app/login-page.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          },
+          body: JSON.stringify({ email, password}),
```
Comment (EtienneDx) [inline packages/front-end/src/app/login-page.tsx]: nit
```suggestion
          body: JSON.stringify({ email, password }),
```

File: packages/front-end/src/app/login-page.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        });
+        const data = await response.json();
+        console.log(data);
```
Comment (EtienneDx) [inline packages/front-end/src/app/login-page.tsx]: data should be in the format { token: string }, so the token should be persisted here (local storage or react context)

File: packages/front-end/src/app/register-page.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      try {
+        const authToken = localStorage.getItem('authToken');
+        const response = await fetch('/register', {
+          method: 'POST',
+          headers: { 
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${authToken}`,
```
Comment (EtienneDx) [inline packages/front-end/src/app/register-page.tsx]: Same comment, no token yet

File: packages/front-end/src/app/register-page.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+          },
+          body: JSON.stringify({ email, password}),
+        });
+        const data = await response.json();
+        console.log(data);
```
Comment (EtienneDx) [inline packages/front-end/src/app/register-page.tsx]: Same here

File: packages/front-end/tsconfig.spec.json [hunk 9]
```diff
@@ -16,7 +16,7 @@
     "src/**/*.test.jsx",
     "src/**/*.spec.jsx",
     "src/**/*.d.ts"
-  ],
+, "src/app/familytree.d.tsx"  ],
```
Comment (EtienneDx) [inline packages/front-end/tsconfig.spec.json]:
```suggestion
, "src/**/*.d.tsx"  ],
```
should give the same result but more consistency

**Review by EtienneDx — COMMENTED** [review]
I pointed out a couple things. Another thing that would be nice is making file names uniform. Otherwise, all seems good

File: packages/front-end/src/app/FamilyTreeBuilder.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+type FamilyTreeProps = {
+  peoples: Person[];
+};
+
+const FamilyTreeBuilder = ({ peoples }: FamilyTreeProps) => {
+  return (
+    <div style={{ height: '100%' }}>
+      <FamilyTree nodes={peoples} />
```
Comment (Lerri-Cofannos) [inline packages/front-end/src/app/FamilyTreeBuilder.tsx]: 'peoples' makes my eyes bleed, there is an s to be removed please !

**Review by Lerri-Cofannos — APPROVED** [review]

**Review by EtienneDx — APPROVED** [review]
All good!

**Pull request merged**